github/open_close_pr.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python3
+
+import os
+import github3
+import argparse
+import logging
+import yaml
+from cachecontrol.adapter import CacheControlAdapter
+import git
+from git import Repo
+
+
+REF_FORMAT = 'refs/heads/'
+DEFAULT_COMMIT_MSG = 'automated commit'
+FREEIPA_PRCI_CONFIG_FILE = '.freeipa-pr-ci.yaml'
+REPO_FORK_REMOTE_NAME = 'mygithub'
+REMOTE_REPO = 'https://github.com/freeipa/freeipa.git'
+UPSTREAM_REMOTE_REF = 'upstream'
+MYGITHUB_REMOTE_REF = 'mygithub'
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+consoleHandler = logging.StreamHandler()
+consoleHandler.setLevel(logging.INFO)
+logger.addHandler(consoleHandler)
+
+
+def load_yaml(yml_path):
+    try:
+        with open(yml_path) as yml_file:
+            return yaml.load(yml_file)
+    except IOError as exc:
+        raise argparse.ArgumentTypeError(
+            'Failed to open {}: {}'.format(yml_path, exc))
+    except yaml.YAMLError as exc:
+        raise argparse.ArgumentTypeError(
+            'Failed to parse YAML from {}: {}'.format(yml_path, exc))
+
+
+class AutomatedPR(object):
+
+    def __init__(self, github_token, repo):
+        github = github3.login(token=github_token)
+        github.session.mount('https://api.github.com', CacheControlAdapter())
+        self.repo = github.repository(repo['owner'], repo['name'])
+
+    def commit_new_prci_config_file(self, new_prci_config, args):
+        """
+        Updates the .freeipa-pr-ci.yaml file with the content
+        of the --prci_config provided file.
+        """
+        prci_config_file = os.path.join(args.repo_path,
+                                        FREEIPA_PRCI_CONFIG_FILE)
+
+        content = ''
+        with open(new_prci_config, 'r') as f:
+            content = f.read()
+
+        with open(prci_config_file, 'w') as f:
+            f.write(content)
+
+        repo = Repo(args.repo_path)
+        repo.git.add(FREEIPA_PRCI_CONFIG_FILE)
+        repo.git.commit('-m', DEFAULT_COMMIT_MSG)
+        repo.git.push(MYGITHUB_REMOTE_REF, args.id)
+
+    def close_older_pr(self, pr_identifier):
+        # repo.git.push('-d', MYGITHUB_REMOTE_REF, pr_identifier)
+        # repo.git.branch('-D', identifier)
+
+        refs = {r.ref:r for r in self.repo.refs()}
```

Comment:
The code below could be replaced by the commented code.

Reply:
I believe it'd be more reliable to just delete the branch based on the identifier. I don't see any benefit in doing it by searching the PRs, just downsides (PR is missing or closed already for some reason; it depletes API requests; ...)